### PR TITLE
collocations

### PR DESCRIPTION
**Pull request by mike-allaway**
Output collocations for monolingal and parallel

### CODE REVIEW

File: django/researchdata/views.py [hunk 1]
```diff
@@ -145,18 +144,15 @@ def get_context_data(self, **kwargs):
...
-                    'frequency': self.request.GET.get('collocations-frequency', ''),
+                    'countby': self.request.GET.get('collocations-countby', 'word'),
+                    'spanleft': int(self.request.GET.get('collocations-spanleft', '3')),
+                    'spanright': int(self.request.GET.get('collocations-spanright', '3')),
+                    'threshold': int(self.request.GET.get('collocations-frequencythreshold', '2')),
+                    'chosen_stats': json.loads(self.request.GET.get('collocations-chosen-stats', '[]')),
+                    # 'sort': 1,
+                    'primlang': primary_language_code
                 }
+                print(options['chosen_stats'])
```
Comment (branfosj) [inline django/researchdata/views.py]:
```suggestion
```

Debug `print` statement?

File: django/researchdata/cwb_output_collocations.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+        except IndexError:
+            continue
+        words = list(set(words))
+        for word in words:
+            if not word:
+                continue
+            if word.lower() == 'that/in':
+                word = 'that'
+            if case_sensitive:
+                word = word.lower()
```
Comment (branfosj) [inline django/researchdata/cwb_output_collocations.py]:
```suggestion
            if not case_sensitive:
                word = word.lower()
```

This looks wrong.

File: django/researchdata/cwb_output_collocations.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    # Case sensitivity
+    case_sensitive = True if '%c' in cwb_query else False  # Case sensitivity
+
+    # Frequency
+    module_dir = os.path.dirname(__file__)
+    freq_path = os.path.join(module_dir, 'freq_birm_eng_word.txt')
+    freq, N = loadFreq(freq_path, case_sensitive)
+
+    # Build collocates dict
+    collocates = {}
```
Comment (branfosj) [inline django/researchdata/cwb_output_collocations.py]: https://docs.python.org/3/library/collections.html#defaultdict-objects

File: django/researchdata/cwb_output_collocations.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+            if not word:
+                continue
+            if word.lower() == 'that/in':
+                word = 'that'
+            if case_sensitive:
+                word = word.lower()
+            if word in collocates:
+                collocates[word] += 1
+            else:
+                collocates[word] = 1
```
Comment (branfosj) [inline django/researchdata/cwb_output_collocations.py]: See the `defaultdict` comment above to simplify this.

**Review by branfosj — CHANGES_REQUESTED** [review]

File: django/researchdata/cwb_output_collocations.py (hunk 2 above)
Comment (mike-allaway) [reply django/researchdata/cwb_output_collocations.py]: In the old code the variable was 'cs' which I expanded out to 'case_sensitive' for clarity. But yes, this should be labelled case_insensitive instead. The value of the variable is correct, just the name, so will disregard the suggested change and update the variable name throughout instead. Thanks for spotting.

**Review by branfosj — APPROVED** [review]

**Pull request merged**